fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -299,30 +311,60 @@ public void set(byte[] key, byte[] value) {
             });
         }
 
+        private void ensureOpen() {
+            if (closed) {
+                throw new RecordCoreStorageException("Agile context is already closed");
+            }
+        }
+
         @Override
         public void flush() {
             commitNow();
             logSelf("Flushed agility context");
         }
+
+        @Override
+        public void flushAndClose() {
+            closed = true;
+            commitNow();
+            logSelf("flushAndClose agility context");
+        }
+
+        @Override
+        public void abortAndReset() {
+            // Here: the lock status is undefined. The main goal of this function is to revive this object for post failure cleanups
+            lock.tryUnlockWrite();
+            boolean releasedLock = lock.tryUnlockRead();
+            for (int maxTries = 20; releasedLock && maxTries > 0; maxTries --) {
+                releasedLock = lock.tryUnlockRead();
+            }
+            committingNow = true;
+            currentContext = null;
```

Comment:
This should call `currentContext.close()` if it is not `null`

Reply:
I wasn't sure about it. Can it be closed after commit failure? 
(meanwhile, ✅ )

Reply:
Yes, you are *always* supposed close contexts. They are tied to native state in the fdb client, that needs to get cleaned up. The close does not commit anything.